atecc608a_se.c

```diff
@@ -185,103 +185,100 @@ static psa_status_t atecc608a_export_public_key(psa_key_slot_number_t key,
     atecc608a_deinit();
     return status;
 }
-static psa_status_t atecc608a_import_public_key(psa_key_slot_number_t key_slot,
-                                                psa_key_lifetime_t lifetime,
-                                                psa_key_type_t type,
-                                                psa_algorithm_t alg,
-                                                psa_key_usage_t usage,
-                                                const uint8_t *p_data,
-                                                size_t data_length)
+static psa_status_t atecc608a_import_public_key(
+    psa_drv_se_context_t *drv_context,
+    psa_key_slot_number_t key_slot,
+    const psa_key_attributes_t *attributes,
+    const uint8_t *data,
+    size_t data_length,
+    size_t *bits)
 {
     const uint16_t key_id = key_slot;
     psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    psa_key_type_t type = psa_get_key_type(attributes);
+    psa_algorithm_t alg = psa_get_key_algorithm(attributes);
 
     ASSERT_SUCCESS_PSA(is_public_key_slot(key_id));
 
     /* Check if the key has a size of 65 {0x04, X, Y}. */
     if (data_length != PSA_KEY_EXPORT_MAX_SIZE(PSA_KEY_TYPE_ECC_PUBLIC_KEY(
                                                    PSA_ECC_CURVE_SECP256R1),
-                                               256))
-    {
+                                               256)) {
         return PSA_ERROR_INVALID_ARGUMENT;
     }
 
-    if (type != PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_CURVE_SECP256R1))
-    {
+    if (type != PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_CURVE_SECP256R1)) {
```

Comment:
Cosmetic: it would make more sense to check the type first, then check that the data length is correct for the sole supported type.